src/backend/trace.rs

```diff
@@ -336,6 +374,18 @@ impl TraceData {
                 let dur = probe.duration();
                 let dur_ms = dur.as_secs_f64() * 1000_f64;
                 hop.total_time += dur;
+                //Before last is set use it to set jitter
+                let last_ms = hop.last_ms().map_or(0f64, |d|d);
+                let jitter_ms = (last_ms - dur_ms).abs();
+                let jitter_dur = Duration::from_millis(jitter_ms as u64);
+                hop.jitter = Some(jitter_dur);
+                let oldjavg = hop.javg;
+                //Welfords online alogrithm to get avg without all values.
+                hop.javg += (jitter_ms - oldjavg) / hop.total_recv as f64;
```

Comment:
should be safe to do this and avoid needing `oldjavg`:

```
 hop.javg += (jitter_ms - hop.javg) / hop.total_recv as f64;
```

Reply:
let mut javg_ms = hop.javg_ms().unwrap_or_default();
                //Welfords online algorithm avg without dataset values.
                javg_ms += (jitter_ms - javg_ms) / hop.total_recv as f64;
                hop.javg = Some(javg_ms);

Agreed it is safe however a little extra work is required because it's a cumulative average(+=). See above.